crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -8,59 +132,67 @@ use std::ops::Range;
 #[derive(Debug, PartialEq, Eq)]
 pub(crate) struct UseDefMap<'db> {
     // TODO store constraints with definitions for type narrowing
+    /// Definition IDs array for `definitions_by_use` and `public_definitions` to slice into.
     all_definitions: Vec<Definition<'db>>,
```

Comment:
Just for myself: Another way to think about `all_definitions` is that it is the definitions arena.